File: app/src/main/java/bj/drunkmessengerblocker/AppComponent.java
package bj.drunkmessengerblocker;

import android.content.Context;

import javax.inject.Singleton;

import bj.drunkmessengerblocker.utils.SharedPrefsManager;
import dagger.Component;

/**
 * Created by j on 15/01/2017.
 */

@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent
{
    void inject(App app);

    Context getApplicationContext();

    SharedPrefsManager getSharedPrefsManager();
}

File: app/src/main/java/bj/drunkmessengerblocker/AppComponent.kt
package bj.drunkmessengerblocker

import android.content.Context
import bj.drunkmessengerblocker.utils.SharedPrefsManager
import dagger.Component
import javax.inject.Singleton

/**
 * Created by Josh Laird on 23/05/2017.
 */
@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {
    fun inject(app: App)

    fun getContext(): Context

    fun getSharedPrefs(): SharedPrefsManager
}